### PR TITLE
Dm 2782

### PR DESCRIPTION
**Pull request by bradjohnson92008**
### JIRA issue link
https://agile6.atlassian.net/browse/DM-2782

## Description - what does this code do?
This codes adds the "Other" option for users to select when they make a comment on a Practice Show page..  the other two options are still available.. i.e., "I am currently adopting this practice" and "I am a member of this practice team".

## Testing done - how did you test it/steps on how can another person can test it 
1.  Log in as a VA (ntlm) user.
2. Navigate to a Practice show page.
3. Scroll down to the comments section and ensure the "Other" option is displayed and selectable..
4. Select the "Other" option - and leave a comment.
5. Click on "Post" button.
6. Ensure the comment displays and in the header of the comment no Banner for "Practice Adopter" or "Team Member" displays.
image
7.  Select "Other and leave replies to the above comment and ensure that the comment displays with no Banner for 'Practice Adopter' or "Team Member".

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Unit tests written (if applicable)
- [ ] e2e/accessibility tests written (if applicable)
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating JIRA issue
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -968,6 +972,7 @@
     t.string "highlight_body"
     t.boolean "retired", default: false, null: false
     t.string "retired_reason"
+    t.boolean "is_public", default: false
```
Comment (emyl3) [inline db/schema.rb:975]: can we remove this change as well?

File: app/controllers/commontator/comments_controller.rb [hunk 2]
```diff
@@ -56,10 +56,9 @@ def create
           user_practice = UserPractice.find_or_create_by(practice_id: @comment.thread.commontable_id, user: @commontator_user)
           @practice = Practice.find_by_id(@comment.thread.commontable_id)
           params[:refresh_page] = user_practice.verified_implementer != (params[:user_practice_status] == 'verified_implementer')
-
-          user_practice.update_attributes(verified_implementer: true, team_member: false) if params[:user_practice_status] == 'verified_implementer'
-          user_practice.update_attributes(verified_implementer: false, team_member: true) if params[:user_practice_status] == 'team_member'
-
+          user_practice.update_attributes(verified_implementer: true, team_member: false, other: false) if params[:user_practice_status] == 'verified_implementer'
+          user_practice.update_attributes(verified_implementer: false, team_member: true, other: false) if params[:user_practice_status] == 'team_member'
+          user_practice.update_attributes(verified_implementer: false, team_member: false, other: true) if params[:user_practice_status] == 'other'
```
Comment (emyl3) [inline app/controllers/commontator/comments_controller.rb:61]: if they don't select any of the options should we default them to "other" or leave the behavior as-is?

**Review by emyl3 — CHANGES_REQUESTED** [review]
Great work, Brad. Have a question and some cleanup stuff in the schema :)

File: .circleci/config.yml [hunk 3]
```diff
@@ -72,7 +72,7 @@ jobs:
           # Bundler 2.2.0 fails to install libv8
           # https://github.com/rubyjs/libv8/issues/310
           command: |
-            sudo apt-get update
+            sudo apt-get update --allow-releaseinfo-change
```
[the remark this answers is not in the document]
Comment (bradjohnson92008) [reply .circleci/config.yml:75]: Had to make change for tests to pass... I could have waited til Josh's PR was in then merged master back in to this branch.

File: db/schema.rb [hunk 4]
```diff
@@ -216,6 +216,10 @@
     t.datetime "updated_at", null: false
   end
 
+  create_table "clinical_id", id: false, force: :cascade do |t|
```
[the remark this answers is not in the document]
Comment (bradjohnson92008) [reply db/schema.rb:219]: Removed th create_table "clinical_id"... method.

File: db/schema.rb (hunk 1 above)
Comment (bradjohnson92008) [reply db/schema.rb:975]: Removed the is_public flag.

**Review by JoshingYou1 — CHANGES_REQUESTED** [review]
Everything looked great except for one thing Brad! Nice work! When I clicked on the edit icon, the practice team member radio option turned into the "other" option. 

Screen Shot 2021-08-17 at 2 27 24 PM

File: app/controllers/commontator/comments_controller.rb (hunk 2 above)
Comment (bradjohnson92008) [reply app/controllers/commontator/comments_controller.rb:61]: THis data is derived from the user_practices table... if you've never commented on a practice.. no radio buttons are selected... if you select a radio button and comment... the app will remember your "state" and pre-populate the same radio button if you make another comment.  You don't have to select an option if you don't want to.   
image

**Comment by bradjohnson92008** [comment]
Good Catch Josh - I was displaying "Other" twice in the "Edit" scenario.. this has been addressed.

**Review by emyl3 — APPROVED** [review]
looks good to me! thanks, Brad!!

**Review by JoshingYou1 — APPROVED** [review]
Thank you for making that change Brad, everything works as intended now. Nice work!

**Pull request merged**